feat: add `fs/append-file`

HRIDYANSHU054 opened the pull request:
adds file system append file utility to Stdlib

## Description

> What is the purpose of this pull request?
This pull request brings the node js file system append-file utility to Stdlib

This pull request:

is aligned with the purpose of achievinng feature parity with Node.js fs package. It Brings file system's powerful append file utility to Stdlib expanding its existing fs utilities.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

Implementation details

-- async version
```
var join = require("path").join;

var fpath = join(__dirname, "examples", "fixtures", "file.txt");

appendFile(fpath, "beep boop\n", onAppend);

function onAppend(error) {
  if (error) {
    throw error;
  }
}
```

-- sync version
```
var appendFile = require( '@stdlib/fs/append-file' );

// Explicitly handle the error...
var err = appendFile( './beep/boop.txt', 'data to append\n' );
if ( err instanceof Error ) {
     throw err;
}
```

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

HRIDYANSHU054 committed:
added append-file utility in stdlib/fs

kgryte commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

kgryte commented:
Thanks, @HRIDYANSHU054, for working on this. This PR is currently failing CI. Those failures need to be addressed before the PR can be reviewed.

HRIDYANSHU054 commented:
Hi @kgryte , Thanks for checking this PR. I m currently working on solving these issues. Majority of them have arised because the fixtures directory was somehow not pushed to github.

HRIDYANSHU054 committed:
Attempts to solve the following issues in the PR

-> run_affected_benchmarks / Run affected benchmarks (pull_request)
-> run_affected_examples / Run changed examples (pull_request)
-> run_affected_tests / Run affected tests (pull_request)

HRIDYANSHU054 committed:
Merge branch 'added-fs/appendFile' of https://github.com/HRIDYANSHU054/stdlib into added-fs/appendFile

HRIDYANSHU054 committed:
attempt to solve the linting issues cming in the previous PRa

HRIDYANSHU054 commented:
@kgryte can you please have a look, I have made changes to tackle the previous PR issues. Though this might seem a silly question but why is it stuck in "Expected — Waiting for status to be reported" , is there any workaround I could do to get it cleared

HRIDYANSHU054 committed:
Made changes to adhere to the repo lint rules and solve othr errors highlighted during the CI check

HRIDYANSHU054 commented:
Corrected some more things. @kgryte can you once again check this

HRIDYANSHU054 commented:
@kgryte can you please approve the remaining workflows. I am interested in this and would like to tackle any other issue that arises or any change that is required.

HRIDYANSHU054 committed:
Solves some more ESlint and JSdoc issues

- error  Begin description with `Append` instead of `Appends`  stdlib/jsdoc-main-export
- Expected space(s) after "catch".   keyword-spacing
- Unexpected sync method: 'appendFileSync'.   node/no-sync

HRIDYANSHU054 commented:
@kgryte all the issues that have been highlighted thus far during the CI checks have been solved. Can you again approve the remaining workflows so that if there are any other errors then they could also be solved and this pr can thus be reviewed. On a sidenote really thanks for your patience✨.

HRIDYANSHU054 committed:
Solved linting and JSDoc issues

- error  Unnecessary try/catch wrapper  no-useless-catch
- error  Missing JSDoc comment  require-jsdoc

HRIDYANSHU054 commented:
@kgryte can you please have a look

HRIDYANSHU054 committed:
Imporoved some more CI check errors

HRIDYANSHU054 commented:
@kgryte please check

HRIDYANSHU054 committed:
Attempted to solve the errors coming during CI

HRIDYANSHU054 commented:
@kgryte please rerun the CI

HRIDYANSHU054 committed:
Corrected JSdoc issues

HRIDYANSHU054 committed:
Solved some more JSdoc errors

HRIDYANSHU054 commented:
@kgryte please can you again check it

HRIDYANSHU054 committed:
some more CI checks solved

HRIDYANSHU054 commented:
@kgryte please check these changes

HRIDYANSHU054 committed:
Solved require statement of file is missing a file extension.

HRIDYANSHU054 commented:
@kgryte please have a look , a Lot of CI check issues have been solved and now only the lint check is remaining to be solved

HRIDYANSHU054 committed:
Solves -> Block-scoped variables in strict mode are not supported until Node.js 4.0.0.

HRIDYANSHU054 commented:
@kgryte please check

HRIDYANSHU054 commented:
Finally, this has passed all checks. @kgryte please can you review it now.

HRIDYANSHU054 commented:
@kgryte this is now clear for review. Kindly check

HRIDYANSHU054 commented:
@kgryte please review my pr

HRIDYANSHU054 commented:
@kgryte the errors that were coming during the CI checks have all been solved. Please review this PR and tell me if any change is needed.

kgryte commented:
@HRIDYANSHU054 
Would you mind checking the box in the OP indicating that you've read the contributing guidelines? This needs to be checked in order to confirm that you agree to licensing terms and the Developer's Certificate of Origin.

kgryte commented on lib/node_modules/@stdlib/fs/append-file/README.md:
```diff
@@ -0,0 +1,255 @@
+
+## Notes
+
+<!-- run-disable -->
```
I suggest basing this note much closer to https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/fs/read-file#notes. One thing to note is that we do not disable ESLint rules within the examples themselves. To disable lint rules, you need to do `<!-- eslint-disable node/no-sync -->` before the codeblock.

HRIDYANSHU054 replied:
Noted and fixed

kgryte commented on lib/node_modules/@stdlib/fs/append-file/README.md:
```diff
@@ -0,0 +1,255 @@
+var fpath = join( __dirname, 'examples', 'fixtures', 'file.txt' );
+
+// Synchronously append data to a file:
+try {
```
Why are we wrapping in a try/catch when we know the function won't throw errors?

HRIDYANSHU054 replied:
fixed, now in line with the new sync implementation

kgryte commented on lib/node_modules/@stdlib/fs/append-file/README.md:
```diff
@@ -0,0 +1,255 @@
+<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->
+
+<section class="related">
+
+* * *
+
+## See Also
+
+-   <span class="package-name">[`@stdlib/fs/exists`][@stdlib/fs/exists]</span><span class="delimiter">: </span><span class="description">test whether a path exists on the filesystem.</span>
+-   <span class="package-name">[`@stdlib/fs/read-file`][@stdlib/fs/read-file]</span><span class="delimiter">: </span><span class="description">read the entire contents of a file.</span>
```
```suggestion
```

As stated in the comment, don't manually edit this section.

HRIDYANSHU054 replied:
fixed, now this should be pre populated

kgryte commented on lib/node_modules/@stdlib/fs/append-file/README.md:
```diff
@@ -0,0 +1,255 @@
+[@stdlib/buffer/ctor]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/buffer/ctor
+[standard-stream]: https://en.wikipedia.org/wiki/Pipeline_%28Unix%29
+
+<!-- <related-links> -->
+
+[@stdlib/fs/exists]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/fs/exists
+[@stdlib/fs/read-file]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/fs/read-file
+
+<!-- </related-links> -->
+
```
```suggestion
```

HRIDYANSHU054 replied:
fixed

kgryte commented on lib/node_modules/@stdlib/fs/append-file/README.md:
```diff
@@ -0,0 +1,255 @@
+
+<section class="links">
+
+[node-fs]: https://nodejs.org/api/fs.html
```
Separate with line breaks as we do everywhere else in the project.

HRIDYANSHU054 replied:
fixed

kgryte commented on lib/node_modules/@stdlib/fs/append-file/benchmark/benchmark.js:
```diff
@@ -0,0 +1,82 @@
+	b.toc();
+	if ( out instanceof Error ) {
+		b.fail( 'something went wrong' );
+	} else {
+		b.pass( 'benchmark finished' );
+	}
```
```suggestion
	}
	b.pass( 'benchmark finished' );
```

The benchmark should state that it finished regardless of whether an error happened.

HRIDYANSHU054 replied:
fixed

kgryte commented on lib/node_modules/@stdlib/fs/append-file/benchmark/benchmark.js:
```diff
@@ -0,0 +1,82 @@
+	function next() {
+		i += 1;
+		if ( i <= b.iterations ) {
+			return appendFile( FILEPATH, DATA, done );
```
Wait...are you repeatedly appending the file over and over again? If I run this benchmark multiple times, this could lead to a huge file. You need to create a fresh `tmp` file for each invocation of `bench`, but before `t.tic()` and you need to delete that tmp file after `t.toc()`. Otherwise, this could lead to a massive amount of used resources.

kgryte commented on lib/node_modules/@stdlib/fs/append-file/benchmark/benchmark.js:
```diff
@@ -0,0 +1,82 @@
+
+	b.tic();
+	for (i = 0; i < b.iterations; i++) {
+		out = appendFile.sync( FILEPATH, DATA );
```
Same comment.

HRIDYANSHU054 replied:
fixed, but implementation strongly needs review

kgryte commented on lib/node_modules/@stdlib/fs/append-file/bin/cli:
```diff
@@ -0,0 +1,110 @@
+
+	// Create a command-line interface:
+	cli = new CLI({
+		'pkg': require( '@stdlib/fs/append-file/package.json' ),
+		'options': require( '@stdlib/fs/append-file/etc/cli_opts.json' ),
```
Use relative paths: https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/fs/read-file/bin/cli#L48.

HRIDYANSHU054 replied:
relative paths added

kgryte commented on lib/node_modules/@stdlib/fs/append-file/bin/cli:
```diff
@@ -0,0 +1,110 @@
+		opts.flag = flags.flag;
+	}
+	if ( flags.mode ) {
+		opts.mode = parseInt(flags.mode, 8);
```
Spacing.

HRIDYANSHU054 replied:
fixed

kgryte commented on lib/node_modules/@stdlib/fs/append-file/docs/repl.txt:
```diff
@@ -0,0 +1,65 @@
+        Options. If a string, the value is the encoding.
+
+    options.encoding: string|null (optional)
+        Encoding. The encoding option is ignored if the data argument is a
```
Why would the encoding option be ignored?

HRIDYANSHU054 replied:
yes, there is no need at all for this comment "The encoding option is ignored if the data argument is a buffer" in append-file

HRIDYANSHU054 replied:
fixed, by removing the irrelevant comment

kgryte commented on lib/node_modules/@stdlib/fs/append-file/docs/repl.txt:
```diff
@@ -0,0 +1,65 @@
+        Options. If a string, the value is the encoding.
+
+    options.encoding: string|null (optional)
+        Encoding. The encoding option is ignored if the data argument is a
```
Same comment.

HRIDYANSHU054 replied:
fixed

kgryte commented on lib/node_modules/@stdlib/fs/append-file/docs/types/index.d.ts:
```diff
@@ -0,0 +1,133 @@
+*/
+interface Options {
+	/**
+	* Encoding; ignored if the data argument is a buffer (default: null).
```
Same comment.

HRIDYANSHU054 replied:
fixed

kgryte commented on lib/node_modules/@stdlib/fs/append-file/docs/types/index.d.ts:
```diff
@@ -0,0 +1,133 @@
+	*
+	* @param path - file path or file descriptor
+	* @param data - data to append
+	* @param options - options; if a string, the value is the encoding
```
```suggestion
	* @param options - options (if a string, the value is the encoding)
```

HRIDYANSHU054 replied:
fixed

kgryte commented on lib/node_modules/@stdlib/fs/append-file/docs/types/index.d.ts:
```diff
@@ -0,0 +1,133 @@
+	*
+	* @param path - file path or file descriptor
+	* @param data - data to append
+	* @param options - options; if a string, the value is the encoding
```
```suggestion
	* @param options - options (if a string, the value is the encoding)
```

HRIDYANSHU054 replied:
fixed

kgryte commented on lib/node_modules/@stdlib/fs/append-file/docs/types/index.d.ts:
```diff
@@ -0,0 +1,133 @@
+	* @param options - options; if a string, the value is the encoding
+	*
+	* @example
+	* try {
```
Why are you wrapping?

Also, within JSDoc comments, you need to space indent, not TAB indent.

HRIDYANSHU054 replied:
fixed

kgryte commented on lib/node_modules/@stdlib/fs/append-file/docs/types/index.d.ts:
```diff
@@ -0,0 +1,133 @@
+	*	   throw err;
+	*  }
+	*/
+	sync( path: string | Buffer | number, data: string | Buffer, options?: Options | string ): void;
```
It should return an `Error` or `null`.

HRIDYANSHU054 replied:
fixed, now in line with the new sync implementation

kgryte commented on lib/node_modules/@stdlib/fs/append-file/docs/types/index.d.ts:
```diff
@@ -0,0 +1,133 @@
+*
+* @param path - file path or file descriptor
+* @param data - data to append
+* @param options - options; if a string, the value is the encoding
```
Same comment.

HRIDYANSHU054 replied:
fixed

kgryte commented on lib/node_modules/@stdlib/fs/append-file/docs/types/index.d.ts:
```diff
@@ -0,0 +1,133 @@
+*/
+declare var appendFile: AppendFile;
+
+// EXPORTS //
```
```suggestion

// EXPORTS //
```

HRIDYANSHU054 replied:
done

kgryte commented on lib/node_modules/@stdlib/fs/append-file/docs/types/test.ts:
```diff
@@ -0,0 +1,219 @@
+	appendFile( '/path/to/beepboop', 'beepboop', 123, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', [], onAppend ); // $ExpectError
+	appendFile( 
+		'/path/to/beepboop',
+		'beepboop',
+		( x: number ): number => x,
+		onAppend
```
Don't do this. Nowhere in the codebase do we split arguments over separate lines.

HRIDYANSHU054 replied:
fixed

kgryte commented on lib/node_modules/@stdlib/fs/append-file/docs/types/test.ts:
```diff
@@ -0,0 +1,219 @@
+	appendFile( '/path/to/beepboop', 'beepboop', { encoding: false }, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', { encoding: [] }, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', { encoding: {} }, onAppend ); // $ExpectError
+	appendFile(
```
Same comment.

HRIDYANSHU054 replied:
fixed

kgryte commented on lib/node_modules/@stdlib/fs/append-file/docs/types/test.ts:
```diff
@@ -0,0 +1,219 @@
+	appendFile( '/path/to/beepboop', 'beepboop', { flag: [] }, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', { flag: {} }, onAppend ); // $ExpectError
+	appendFile(
+		'/path/to/beepboop',
```
Same comment here and everywhere else.

HRIDYANSHU054 replied:
fixed

kgryte commented on lib/node_modules/@stdlib/fs/append-file/docs/types/test.ts:
```diff
@@ -0,0 +1,219 @@
+
+// Attached to main export is a `sync` method which returns an error or null...
+{
+	appendFile.sync( '/path/to/beepboop', 'beepboop' ); // $ExpectType Error | null
```
@Planeshifter For some reason, linting on CI did not pick up this failure. The `index.d.ts` is incorrectly typed, while this assertion is correctly typed. I would have expected a CI failure.

Planeshifter replied:
@kgryte Typing in `index.d.ts` matches the assertion here now, so I consider this resolved.

kgryte commented on lib/node_modules/@stdlib/fs/append-file/docs/types/test.ts:
```diff
@@ -0,0 +1,219 @@
+	appendFile.sync( '/path/to/beepboop', 'beepboop', { flag: [] } ); // $ExpectError
+	appendFile.sync( '/path/to/beepboop', 'beepboop', { flag: {} } ); // $ExpectError
+	appendFile.sync( '/path/to/beepboop', 'beepboop', {
+		flag: (x: number): number => x,
```
Same thing. We don't split over lines like this in test.ts files. As the saying goes, "when in Rome..."

HRIDYANSHU054 replied:
fixed, this splitting was a result of the trailing comma after the last property in the object.

kgryte commented on lib/node_modules/@stdlib/fs/append-file/examples/index.js:
```diff
@@ -0,0 +1,42 @@
+var fpath = join( __dirname, 'fixtures', 'file.txt' );
+
+// Synchronously append data to a file:
+try {
```
Same comments as README.

HRIDYANSHU054 replied:
fixed

kgryte commented on lib/node_modules/@stdlib/fs/append-file/lib/sync.js:
```diff
@@ -0,0 +1,59 @@
+			appendFile( path, data );
+		}
+	} catch ( err ) {
+		console.log( err.message );
```
```suggestion
```

Never leave console statements lying around in implementations.

HRIDYANSHU054 replied:
fixed, now in line with the new sync implementation

kgryte commented on lib/node_modules/@stdlib/fs/append-file/lib/sync.js:
```diff
@@ -0,0 +1,59 @@
+		}
+	} catch ( err ) {
+		console.log( err.message );
+		throw err;
```
Why are you throwing? One of the specific rationales for our `fs` packages is the ability to explicitly handle errors by returning as return values. See https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/fs/read-file/lib/sync.js#L53.

HRIDYANSHU054 replied:
since, this was implemented a while back, I took the implementation notes from Node fs documentaion, this is even visible in the first commit message 
The error handling approach is also different from stdlib's approach
```
-- sync version
var appendFile = require( '@stdlib/fs/append-file' );

// Explicitly handle the error...
try{
appendFile.sync( '/path/to/file.txt', 'beep boop\n' );
} catch(err) {
throw err
}
```

in accordance to stdlib
```
var err = appendFile.sync( '/path/to/file.txt', 'beep boop\n' );
if (err) {
throw err;
}
```
now when I ve seen and have gone through some of the stdlib implementations YES this needs to return the error and n ot throw it. and I will be working on this as soon as possible.

kgryte reviewed (changes requested):
@HRIDYANSHU054 I've left a number of initial comments. This implementation needs a fair amount of clean-up before it can be ready for merge.

HRIDYANSHU054 commented:
@kgryte thanks for going through this and reviewing it. I don't why the splitting has happened in only some functions and not others even though I have configured my ESlint to highlight these issues if they occur. Also can I address this after 5th april as I am currenlty having fromm today my mid semester examinations.

HRIDYANSHU054 committed:
Code Review Requested changes have implemented

major changes
- approach for sync version chnaged from throwing errors to returing errors.
- approach for benchmarking, now the file is deleted after every itr and then created in the next itr by appendFile

HRIDYANSHU054 committed:
some more index.d.ts style fixes

HRIDYANSHU054 commented:
@kgryte the changes you highlighted have been implemented, certain things like the new benchmarking approach strongly needs to be reviewed

HRIDYANSHU054 commented on lib/node_modules/@stdlib/fs/append-file/benchmark/benchmark.js:
```diff
@@ -0,0 +1,83 @@
+
+	return next();
+
+	function next() {
+		i += 1;
+		if ( i <= b.iterations ) {
+			return appendFile( TMP, DATA, done );
+		}
+		b.toc();
+		unlink( TMP );
+		b.pass( 'benchmark finished' );
+		b.end();
+	}
```
uses capability of append-file to create the file if the file does not exists yet, and then deletes the file after its done to avoid wasting memory resources

HRIDYANSHU054 committed:
fixed some CI issues

HRIDYANSHU054 committed:
fixed some CI issues

HRIDYANSHU054 commented:
@kgryte can you approve these remaining workflows and then review the changes

HRIDYANSHU054 commented:
Hi ✋ @kgryte just hoping you could further review these changes, and anything more that you want me to change or add I will work on that also.

HRIDYANSHU054 commented:
hi @kgryte could you review this

HRIDYANSHU054 commented:
@kgryte could you review this PR

Planeshifter commented on lib/node_modules/@stdlib/fs/append-file/test/fixtures/stdin_error.js.txt:
```diff
@@ -0,0 +1,34 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

Planeshifter commented on lib/node_modules/@stdlib/fs/append-file/test/fixtures/write_error.js.txt:
```diff
@@ -0,0 +1,43 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

Planeshifter commented on lib/node_modules/@stdlib/fs/append-file/package.json:
```diff
@@ -0,0 +1,69 @@
+{
+  "name": "@stdlib/fs/append-file",
+  "version": "0.0.0",
+  "description": "Appends data to a file.",
```
```suggestion
  "description": "Append data to a file.",
```

Planeshifter commented on lib/node_modules/@stdlib/fs/append-file/README.md:
```diff
@@ -0,0 +1,237 @@
+
+// Synchronously append data to a file:
+var error = appendFile.sync( fpath, 'beep boop\n', 'utf8' );
+// Function successfully executes and returns null
```
```suggestion
// returns null
```

Planeshifter reviewed (approved):
Apologies for the late review; thanks for your PR, which looks very good now. I will merge shortly after CI has passed again after some tiny changes.

Planeshifter committed:
Update lib/node_modules/@stdlib/fs/append-file/test/fixtures/stdin_error.js.txt

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
Update lib/node_modules/@stdlib/fs/append-file/README.md

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
Update lib/node_modules/@stdlib/fs/append-file/package.json

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
Update lib/node_modules/@stdlib/fs/append-file/test/fixtures/write_error.js.txt

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/fs/append-file/README.md:
```diff
@@ -0,0 +1,237 @@
+        console.error( 'Error: %s', error.message );
+    }
+    console.log( 'Success!!!' );
+}
```
```suggestion
var join = require( 'path' ).join;
var appendFile = require( '@stdlib/fs/append-file' );

var fpath = join( __dirname, 'examples', 'fixtures', 'file.txt' );

// Synchronously append data to a file:

var error = appendFile.sync( fpath, 'beep boop\n', 'utf8' );
// Function successfully executes and returns null

console.log( error instanceof Error );
// => false

// Asynchronously append data to a file:

appendFile( fpath, 'beep boop\n', onAppend );

function onAppend( error ) {
    if ( error ) {
        console.error( 'Error: %s', error.message );
    }
    console.log( 'Success!!!' );
}
```

Planeshifter commented on lib/node_modules/@stdlib/fs/append-file/README.md:
```diff
@@ -0,0 +1,237 @@
+
+var fpath = join( __dirname, 'examples', 'fixtures', 'file.txt' );
+
+// Synchronously append data to a file:
+var error = appendFile.sync( fpath, 'beep boop\n', 'utf8' );
+// returns null
+
+console.log( error instanceof Error );
+// => false
+
+// Asynchronously append data to a file:
+appendFile( fpath, 'beep boop\n', onAppend );
+
+function onAppend( error ) {
+    if ( error ) {
+        console.error( 'Error: %s', error.message );
+    }
+    console.log( 'Success!!!' );
+}
```
```suggestion
// Synchronously append data to a file:

var error = appendFile.sync( fpath, 'beep boop\n', 'utf8' );
// Function successfully executes and returns null

console.log( error instanceof Error );
// => false

// Asynchronously append data to a file:

appendFile( fpath, 'beep boop\n', onAppend );

function onAppend( error ) {
    if ( error ) {
        console.error( 'Error: %s', error.message );
    }
    console.log( 'Success!!!' );
}
```

Planeshifter committed:
Update lib/node_modules/@stdlib/fs/append-file/README.md

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

HRIDYANSHU054 commented:
Thanks @Planeshifter looking forward to learning and contribute more to stdlib node's environment
